core/src/main/scala/com/banno/kafka/consumer/ConsumerImpl.scala

```diff
@@ -36,35 +37,38 @@ case class ConsumerImpl[F[_], K, V](c: Consumer[K, V])(implicit F: Sync[F])
     F.delay(c.assignment().asScala.toSet)
 
   override def close: F[Unit] =
-    log.debug(s"${Thread.currentThread.getId} consumer.close()...") *> F.delay(
-      c.close()
-    ) *> log
-      .debug(s"${Thread.currentThread.getId} consumer.close()")
+    F.interruptible(c.close())
+
   override def close(timeout: FiniteDuration): F[Unit] =
-    log.debug(s"${Thread.currentThread.getId} consumer.close($timeout)...") *> F
-      .delay(
-        c.close(java.time.Duration.ofMillis(timeout.toMillis))
-      ) *> log.debug(s"${Thread.currentThread.getId} consumer.close($timeout)")
-  override def commitAsync: F[Unit] = F.delay(c.commitAsync())
+    F.interruptible(c.close(JDuration.ofMillis(timeout.toMillis)))
+
+  override def commitAsync: F[Unit] =
+    F.delay(c.commitAsync())
+
   override def commitAsync(
       offsets: Map[TopicPartition, OffsetAndMetadata],
       callback: OffsetCommitCallback,
-  ): F[Unit] = F.delay(c.commitAsync(offsets.asJava, callback))
+  ): F[Unit] =
+    F.delay(c.commitAsync(offsets.asJava, callback))
+
   override def commitAsync(callback: OffsetCommitCallback): F[Unit] =
     F.delay(c.commitAsync(callback))
-  override def commitSync: F[Unit] = F.delay(c.commitSync())
+
+  override def commitSync: F[Unit] =
+    F.interruptible(c.commitSync())
+
   override def commitSync(
       offsets: Map[TopicPartition, OffsetAndMetadata]
   ): F[Unit] =
-    F.delay(c.commitSync(offsets.asJava))
+    F.interruptible(c.commitSync(offsets.asJava))
 
   override def listTopics: F[Map[String, Seq[PartitionInfo]]] =
     F.delay(c.listTopics().asScala.toMap.view.mapValues(_.asScala.toSeq).toMap)
```

Comment:
`listTopics`, too, is blocking, yes?

Reply:
Yep - went ahead and changed most of the other `delay` to `interruptible`.